src/mem/ruby/common/DataBlock.hh

```diff
@@ -96,7 +96,8 @@ class DataBlock
     void setData(PacketPtr pkt);
     void copyPartial(const DataBlock &dblk, int offset, int len);
     void copyPartial(const DataBlock &dblk, const WriteMask &mask);
-    void atomicPartial(const DataBlock & dblk, const WriteMask & mask);
+    void atomicPartial(const DataBlock & dblk, const WriteMask & mask,
+            bool isAtomicNoReturn=false);
```

Comment:
I would actually suggest the opposite for the default.  In other words, do not use atomic log by default. I think that will prevent needing to change any of the other protocols with use atomics besides VIPER.  That might be a bit confusing but if we rename the parameter to "noAtomicLog" or similar that would be clear.

Reply:
Oh, yeah that makes more sense. Will make the change.